### PR TITLE
notes: Implement get_it dependency injection

### PR DESCRIPTION
**Pull request by broodroosterdev**
This PR will add get_it dependency injection

### CODE REVIEW

File: lib/locator.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import 'package:get_it/get_it.dart';
+import 'package:potato_notes/data/dao/note_helper.dart';
+import 'package:potato_notes/internal/app_info.dart';
+import 'package:potato_notes/internal/preferences.dart';
+
+import 'main.dart';
+
+final locator = GetIt.instance;
+
+void setupLocator(){
```
Comment (dvd-bnc) [inline lib/locator.dart]: missing space between () and {

File: lib/locator.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  locator.registerSingleton<Preferences>(Preferences());
+  locator.registerLazySingleton<NoteHelper>(() => NoteHelper(db));
```
Comment (dvd-bnc) [inline lib/locator.dart]: consider using something like database.helper, you can pass db in the setupLocator params

**Comment by broodroosterdev** [comment]
```
Running Gradle task 'assembleRelease'...                                
                                                   
Running Gradle task 'assembleRelease'... Done                     140.7s (!)
✓ Built build/app/outputs/apk/release/app-release.apk (22.6MB).
```

File: lib/routes/note_page.dart [hunk 3]
```diff
@@ -18,13 +18,15 @@ import 'package:potato_notes/internal/app_info.dart';
...
 import 'package:potato_notes/internal/utils.dart';
+import 'package:potato_notes/locator.dart';
 import 'package:potato_notes/routes/note_page_image_gallery.dart';
 import 'package:potato_notes/widget/note_color_selector.dart';
 import 'package:potato_notes/widget/note_toolbar.dart';
 import 'package:potato_notes/widget/note_view_images.dart';
 import 'package:provider/provider.dart';
 import 'package:rich_text_editor/rich_text_editor.dart';
 import 'package:spicy_components/spicy_components.dart';
+import 'package:uuid/uuid.dart';
```
Comment (dvd-bnc) [inline lib/routes/note_page.dart]: unused import

File: pubspec.yaml [hunk 4]
```diff
@@ -39,6 +39,9 @@ dependencies:
       url: https://github.com/namhyun-gu/flutter_rich_text_editor
   spicy_components:
     git: https://github.com/HrX03/SpicyUIComponents
+  get_it:
+  json_annotation:
```
Comment (dvd-bnc) [inline pubspec.yaml]: this doesnt serve any purpose afaik

File: lib/routes/note_page.dart (hunk 3 above)
Comment (broodroosterdev) [reply lib/routes/note_page.dart]: Yeah this made my build error actually. forgot to commit it

**Comment by broodroosterdev** [comment]
I still need to implement the changes outlined in the previous review

**Comment by broodroosterdev** [comment]
Added them in latest commit

**Pull request merged**